### ISSUE TITLE
Container.full files does not work well - RPM dep installation issue

### ISSUE THREAD

**Issue by jscotka** [teemtee/tmt#653]
## EDIT: 
with makefile Dockerfile.mini works well probably `.` at the end fixed the issue

I've wanted to use containers and seems that there are few issues. 
firs one is probably easy to fix, second one (tested also with F32 base image did not helper, so probably some added dep what is not able to install inside container)

## Dockerfile.mini
```
$ podman build --tag tmt -f containers/Dockerfile.mini 
STEP 1: FROM registry.fedoraproject.org/fedora
STEP 2: RUN set -x &&     sed -i '/tsflags=nodocs/d' /etc/dnf/dnf.conf &&     echo "Set disable_coredump false" >> /etc/sudo.conf
--> Using cache 5be20df8c867e3a65b86edae280c863931d626b9eb8552dbb6a1342b044e2bf2
--> 5be20df8c86
STEP 3: RUN set -x &&     dnf install -y dnf-plugins-core &&     dnf copr enable -y psss/tmt &&     dnf install -y tmt &&     dnf autoremove -y &&     dnf clean all
--> Using cache a59e1463bfdef12b6dffc7aa964cff1d3d71d9c9502d01de75407b89ae6ec545
--> a59e1463bfd
STEP 4: RUN mkdir /tmt
--> Using cache 53a742a2d1ed58fc1d9d9cef3ed950725af9abfa74dc0c41d7007c4da746eb03
--> 53a742a2d1e
STEP 5: COPY .fmf /tmt/.fmf
Error: error building at STEP "COPY .fmf /tmt/.fmf": error adding sources [/home/jscotka/git/tmt/containers/.fmf]: error checking on source /home/jscotka/git/tmt/containers/.fmf under "/home/jscotka/git/tmt/containers": copier: stat: "/.fmf": no such file or directory
```
## Dockerfile.full
```
$ podman build --tag tmt -f containers/Dockerfile.full
...

Failed:
  iputils-20200821-1.fc33.x86_64                                                

Error: Transaction failed
Error: error building at STEP "RUN set -x &&     dnf install -y dnf-plugins-core &&     dnf copr enable -y psss/tmt &&     dnf install -y tmt-all beakerlib python3-{mock,test} &&     dnf autoremove -y &&     dnf clean all": error while running runtime: exit status 1
```

**Comment by psss** [comment]
Have you tried `make images`?

**Comment by jscotka** [comment]
@psss as you mentioned, there were issue with location so that `.` at the end of cmd line helper for `Dockerfile.mini`

But `Dockerfile.full` is RPM level issue, so it failed also with this `make images`

**Comment by psss** [comment]
Hm, interesting. The `iputils` packages cannot be installed. Tried installing only  this one package and it failed also with `fedora:32`.

**Comment by jscotka** [comment]
> Hm, interesting. The `iputils` packages cannot be installed. Tried installing only this one package and it failed also with `fedora:32`.

On the one hand it is expected that not all packages could be installed inside container, on the other hand, something added this dependency somehow. Or it is possible that with some older fedora <F32 it worked and it was broken by someone when building RPM packages for fedora

**Comment by jscotka** [comment]
Also tested F30 and also not able to install 
```
Running transaction
  Preparing        :                                                        1/1 
  Installing       : iputils-20180629-4.fc30.x86_64                         1/1Error unpacking rpm package iputils-20180629-4.fc30.x86_64
 
  Verifying        : iputils-20180629-4.fc30.x86_64                         1/1 

Failed:
  iputils-20180629-4.fc30.x86_64                                                

Error: Transaction failed
Error: error building at STEP "RUN dnf install -y iputils": error while running runtime: exit status 1
```

**Comment by jscotka** [comment]
so this dependecy is part of `tmt-all` package

**Comment by jscotka** [comment]
similar issue inside BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1915377

**Comment by pvalena** [comment]
@jscotka The images for regular usage are already built here: 
  https://quay.io/repository/testing-farm/tmt-all
  https://quay.io/repository/testing-farm/tmt

Also, please note that: F32 support ends on 2021-05-18. Earlier versions are no longer supported.

**Comment by jscotka** [comment]
@pvalena yep, but this issue is more about that you are not able to regenerate image now. I wanted to create new image for test using requre for files regeneration and net this issue. So i dont need it directly. With okder

**Comment by pvalena** [comment]
> @pvalena yep, but this issue is more about that you are not able to regenerate image now. I wanted to create new image for test files regeneration for requre and met this issue. So i dont need it directly.
> 
> With older version of fedora we just tried to find what happened.

I'd aim for the latest stable distro release only (as that's what builds in our quay.io anyways).

```
$ podman run --rm -ti fedora cat /etc/os-release
NAME=Fedora
VERSION="33 (Container Image)"
[ . . . ]
$ podman run -it --rm quay.io/testing-farm/tmt bash -c "cat /etc/os-release"
NAME=Fedora
VERSION="33 (Container Image)"
[ . . . ]
```

Otherwise, we need to end up with double the number of Dockerfiles, as I'd rather avoid having some decision logic inside one.

**Comment by jscotka** [comment]
@pvalena not sure If I've described it well.
I does not care about older distro. The newest one is good enough.

My inspection of older distros were about to find when happening this issue, that it is unable to rebuild images (and tried to find via older releases if there were added some new dependency caused by some package) and finally we found that `tmt-all` brings this dependency on `iputils`

So now it is unable to regenerate `Container.full` image on `quay.io` and will contain old version of TMT until it will be fixed.

**Comment by pvalena** [comment]
> @pvalena not sure If I've described it well.
> I does not care about older distro. The newest one is good enough.

That's my point, the newest one works for me: https://gist.github.com/pvalena/aa7d7d7011c7edc7e8aeb6fc08676ba3

Is there any issue I can reproduce with F33? Or it was fixed already?

> 
> My inspection of older distros were about to find when happening this issue, that it is unable to rebuild images (and tried to find via older releases if there were added some new dependency caused by some package) and finally we found that `tmt-all` brings this dependency on `iputils`
> 
> So now it is unable to regenerate `Container.full` image on `quay.io` and will contain old version of TMT until it will be fixed.

I see, is there any log with build failure?

**Comment by jscotka** [comment]
@pvalena 
yep, possible that they fix it already in iputils package in fedora. So in this case we can close it.

**Comment by psss** [comment]
Image building now works fine and is automated. Here's a recent run:
https://github.com/teemtee/tmt/actions/runs/17987588428/job/51169564205
